packages/core/src/stream-loading/state-manipulator.ts

```diff
@@ -54,22 +54,16 @@ export class StateManipulator {
   ) {}
 
   /**
-   * Applies log to state and returns a new state.  Most callers should set 'cloneInput' to true
-   * so that the original StreamState is not modified.  Some codepaths may set 'cloneInput' to false
-   * if they are confident that the StreamState has already been cloned once within the
-   * StateManipulator and so the input state that was given to the StateManipulator will not be
-   * modified.
+   * Applies log to state and returns a new state.  Note that the input state may be modified,
+   * if you don't want to change the input state, callers should clone it before passing it into
+   * this function.
    */
   async _applyLog<T extends Stream>(
     handler: StreamHandler<T>,
     state: StreamState | null,
     log: AppliableStreamLog,
     throwOnInvalidCommit: boolean,
-    cloneInput: boolean
```

Comment:
decided to get rid of this argument and make the cloneDeep explicit at the call site.  That seemed clearer and more straightforward to understand to me.